### ISSUE TITLE
Replace twisted.python._release.runCommand with subprocess.check_output

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8648]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)| @markrwilliams reported|
|-|-|
|Trac ID|trac#8648|
|Type|enhancement|
|Created|2016-07-21 14:58:34Z|

`twisted.python._release.runCommand` wraps `subprocess.Popen` so that standard out and error end up in the same byte string, and a non-zero exit status raises an exception.

This is nearly equivalent to `subprocess.check_output`, available in Pythons 2.7 and newer.  The only difference is that `runCommand`'s `CommandFailed` exception differentiates between child processes interrupted by a signal and those that simply exited with a non-zero status.  Nothing seems to actually care about that difference, though, so it should be safe to substitute with `subprocess.CalledProcessError`

<details><summary>Searchable metadata</summary>

```
trac-id__8648 8648
type__enhancement enhancement
reporter__markrwilliams markrwilliams
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469113114293277 1469113114293277
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_1"></a>|@markrwilliams commented|
|-|-|

After a bit of thought it seems better to keep {runCommand}, but to change its implementation to use {check_output}.

See https://github.com/twisted/twisted/pull/383

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset fc120b8a9313fa73bbe70c9dc58442f22f44e696
```
#!CommitTicketReference repository="" revision="fc120b8a9313fa73bbe70c9dc58442f22f44e696"
Merge pull request #396 from twisted/8648-markrwilliams-replace-runcommand-with-check_output

Author: markrwilliams
Reviewer: rodrigc
Fixes: #8648

twisted.python._release.runCommand now uses check_output
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
